niworkflows/interfaces/images.py

```diff
@@ -670,25 +670,38 @@ class SignalExtraction(SimpleInterface):
     output_spec = SignalExtractionOutputSpec
 
     def _run_interface(self, runtime):
+        img = nb.load(self.inputs.in_file)
         mask_imgs = [nb.load(fname) for fname in self.inputs.label_files]
-        if len(mask_imgs) == 1:
+        multi_index_mask = False
+        if len(mask_imgs) == 1 and len(mask_imgs[0].shape) == 4:
             mask_imgs = nb.four_to_three(mask_imgs[0])
 
-        masks = [mask_img.get_data().astype(np.bool) for mask_img in mask_imgs]
+        for i, mask in enumerate(mask_imgs):
+            if img.shape != mask.shape:
+                mask_imgs[i] = nli.resample_to_img(
+                    mask, img, interpolation='nearest')
 
-        n_masks = len(masks)
+        if len(mask_imgs) > 1:
+            masks = [mask_img.get_data().astype(np.bool)
+                     for mask_img in mask_imgs]
+            n_masks = len(masks)
+        else:
+            masks = mask_imgs[0].get_data()
```

Comment:
why not preparing everything in this block? And what happens when one hemisphere has positive labels and the other negative?

```Python
            labelsmap = mask_imgs[0].get_data()
            labels = np.unique(labelsmap)
            labels = labels[labels != 0]  # drop background (zero)
            masks = [labelsmap == l for l in labels]
```